engine/src/test/java/com/redhat/cloud/notifications/transformers/BaseTransformerTest.java

```diff
@@ -1,36 +1,114 @@
+        metadata.setAdditionalProperty(FIXTURE_METADATA_ADDITIONAL_PROPERTY, FIXTURE_METADATA_ADDITIONAL_PROPERTY_VALUE);
+        event.setMetadata(metadata);
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        final Payload payload = new Payload();
+        payload.setAdditionalProperty(FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY, FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY_VALUE);
+        event.setPayload(payload);
 
-        assertEquals("someOrgId", jsonObject.getString("org_id"));
+        events.add(event);
+        FIXTURE_EVENTS = events;
     }
 
+    /**
+     * Tests that a proper JSON payload is generated from a fully populated {@link Action}.
+     */
     @Test
-    void shouldNotRaiseAnExceptionWhenOrgIdIsMissing() {
+    void toJsonObjectTest() {
         Action action = new Action();
-        action.setTimestamp(LocalDateTime.of(2022, 12, 24, 12, 0));
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        action.setAccountId(FIXTURE_ACCOUNT_ID);
+        action.setApplication(FIXTURE_APPLICATION);
+        action.setBundle(FIXTURE_BUNDLE);
+        action.setContext(FIXTURE_CONTEXT);
+        action.setEventType(FIXTURE_EVENT_TYPE);
+        action.setEvents(FIXTURE_EVENTS);
+        action.setOrgId(FIXTURE_ORG_ID);
+        action.setTimestamp(FIXTURE_TIMESTAMP);
+
+        // Call the function under test.
+        JsonObject result = this.baseTransformer.toJsonObject(action);
+
+        // Assert the values.
+        assertEquals(FIXTURE_ACCOUNT_ID, result.getString(BaseTransformer.ACCOUNT_ID), "the account id isn't the same");
+        assertEquals(FIXTURE_APPLICATION, result.getString(BaseTransformer.APPLICATION), "the application isn't the same");
+        assertEquals(FIXTURE_BUNDLE, result.getString(BaseTransformer.BUNDLE), "the bundle isn't the same");
+        assertEquals(FIXTURE_EVENT_TYPE, result.getString(BaseTransformer.EVENT_TYPE), "the event type isn't the same");
+
+        // Assert that the keys that contain inner JSON arrays or objects are present.
+        assertTrue(result.containsKey(BaseTransformer.CONTEXT), "the context is missing from the resulting JSON object");
+        assertTrue(result.containsKey(BaseTransformer.EVENTS), "the events is missing from the resulting JSON object");
+        assertTrue(result.containsKey(BaseTransformer.ORG_ID), "the org id is missing from the resulting JSON object");
+
+        final JsonArray events = result.getJsonArray(BaseTransformer.EVENTS);
+        if (events.size() != FIXTURE_EVENTS.size()) {
+            fail("the number of events in the JSON file is different from the test data");
+        }
+
+        final JsonObject event = events.getJsonObject(0);
+        final JsonObject metadata = event.getJsonObject(BaseTransformer.METADATA);
+        assertTrue(metadata.containsKey(FIXTURE_METADATA_ADDITIONAL_PROPERTY), "the metadata object doesn't contain the additional property's key");
```

Comment:
This could be removed since there's a value check below.
```suggestion
```

Reply:
This too!